### DISCUSSION TITLE
Please return compatibility with Automatic1111 1.4 and up (1.5.1 at the moment)

### DISCUSSION THREAD

**Discussion by ZonSoft** [volotat/SD-CN-Animation#187] [General]
Multiple errors after installing this extension. Pitifully not usable any more,

**Comment by vapidvim** [comment]
@ZonSoft I just turn off other extensions when using it and then turn it off when I want to use everything else.. Its worth it when it works. :)

**Reply by ricardofeynman** [reply]
@vapidvim I'm using an older version of A1111 (a strictly SD-CN dedicated A1111 1.2)  as I couldn't find a combination that worked well with earlier commits. Have you been having success with 1.7 and if so, which commit of SD-CN are you using? Thanks. 

Also for those who don't care what extra hoops they need to jump through to get this gem operational, see: https://github.com/volotat/SD-CN-Animation/issues/203#issuecomment-1904203379

**Reply by vapidvim** [reply]
I have it working in version A1111 1.3.2 and I might have made some small revisions to the code to get it working with CPU for Mac M1. I just installed the new A1111 on a different drive.

**Reply by ricardofeynman** [reply]
If you have any luck with 1.7, I'd be very grateful for an update. 

Any version after 1.5.1 that SD-CN plays nice with would be a good starting point for me to test out an idea.

Unfortunately I did observe noteable output quality degredations with later SD-CN commits when retesting the same prompts in t2v. Do you happen to know which commit of SD-CN works for you with 1.3.2?

**Reply by vapidvim** [reply]
Yeah I have been trying to find that.. I went to a previous commit because the most recent version didn't work on my machine. I have one archived from May 30th that was working good and currently have one from Sept/Oct.. I should go back and try that May 30th one again! https://github.com/volotat/SD-CN-Animation/compare/v0.9...main

**Reply by vapidvim** [reply]
https://github.com/volotat/SD-CN-Animation/tree/2e257bbfc3b76fb1323f7b22f820981c1b9f56b2
I think this is the one.. File modified dates stamps I have are May 27 and Jun 1

**Reply by ricardofeynman** [reply]
Thanks I'll test the 6 commits made between May 27 and Jun 1 with 1.3.2. 

When you say you have one from Sept/Oct, are we talking about the SD-CN commit from Sep 1st (the alexbofa PR merge https://github.com/volotat/SD-CN-Animation/tree/ab1f1583b78211c55c401bd21e13cc91747a590e ) in combination with 1.6.0?

**Reply by ricardofeynman** [reply]
> Yeah I have been trying to find that..

Did you mean you've been trying to find the SD-CN May 30th commit that gave you best output results with 1.3.2, or were you talking about the quest for a version that works well with 1.5.1 or above? 

Seeking clarity as although I'm keen to see if there's any combination that improves upon the output of the combi I first mentioned ( https://github.com/volotat/SD-CN-Animation/issues/203#issuecomment-1904203379 ), my main objective* at present is finding any version of SD-CN that loads with any A1111 main/dev branch commit after 1.5.1 was released. 

I can then do a diff check between the scripts in https://github.com/volotat/SD-CN-Animation/commit/eddf1a4c8fb327e4fca05821490f20d97945ac7e and whichever newer commit of SD-CN happens to load without error, and then attempt to revert any changes to FloweR/RAFT/etc that may have caused output quality regressions for the models/prompts/SD-CN params I'm testing with. 

Ideally I'd like to keep any of the code that improved the extension's functionality in more recent SD-CN commits, but which doesn't impact the code relevant to the output. e.g SciKit package not requiring online connection to redownload the first time A1111 is started, which was still an irritation in https://github.com/volotat/SD-CN-Animation/commit/eddf1a4c8fb327e4fca05821490f20d97945ac7e or the settings .json being automatically saved to the frames output folder, etc. 

I could also get by with the limited featureset of the earlier commits, reverting fully except for the necessary code required to work with newer versions of A1111. 

*Perhaps I should be checking which version of A1111 on dev/main was in use when alexbofa made that SD-CN PR...

**Reply by vapidvim** [reply]
I was trying to find any version number information in the extension. I didn't notice a difference in quality, but I wasn't looking for that. I think at some point the newest version didn't work at all and so I went back to that version I linked above. And then at some point I updated the extension again and it worked (must have been a fix) and I haven't loaded up the backup zip I have of the May version that was working before. 
I think our best bet is to plug in the A1111 repository link into a GTP that can analyze repos and then do the same with this extension and troubleshoot the error messages from there. I have a few other projects atm, but if I get to it before someone else I will post the fork here.

**Reply by ricardofeynman** [reply]
If I'm following correctly, I'm pretty sure we can just do a git checkout to establish the exact hash for any extension's commit that you have installed. Will have a refresh on how to do that if that's something you're unsure about, but yes, firing up an LLM for rapid analysis sounds like a good call, I haven't tried that out yet.

I stay clear of chatGPT but earlier this week I saw that there's a local running open source coding LLM that is coming reasonably close in capabilities (from the metrics I've seen anyway) so I'll look up the details of the best OS LLM's available at present for code related tasks. If you already have a greater insight on that front, I'd be grateful for any recommendations.

**Reply by ricardofeynman** [reply]
https://github.com/deepseek-ai/DeepSeek-Coder

Possibly this was the LLM family I had in mind last week. I'm still learning and don't think I could apply that knowledge to the task yet.  Have you had any experience with DeepSeek or other LLM's for code analysis?

**Reply by vapidvim** [reply]
I have played with it before using the playground here: https://chat.deepseek.com/
I think GPT-4 is currently the best bet, or maybe Replit since you can import repos and it has ai built in.

**Reply by ricardofeynman** [reply]
Cool, I've not heard of Replit, I'll go take a look. Thanks
